### PR TITLE
feat(starfish): Add slow samples table to endpoint details page

### PR DESCRIPTION
**Pull request by gggritso**
**e.g.,**
Screenshot 2023-05-16 at 1 37 09 PM

### CODE REVIEW

File: static/app/views/starfish/components/sampleEvents.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    route: 'events',
+    eventView,
+    referrer: 'starfish-transaction-summary-sample-events',
+    limit: 5,
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...sampleEventsEventView.getEventsAPIPayload(location),
+    }),
+  });
```
Comment (DominikB2014) [inline static/app/views/starfish/components/sampleEvents.tsx:86]: Could you clarify the reason why we are passing in the `eventView` here for the first parameters, but using the request payload from the `sampleEventsEventView`?

Could we use the `useDiscoverQuery` hook, and just pass in the `sampleEventsEventView`?

File: static/app/views/starfish/components/sampleEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+type Props = {
+  eventView: EventView;
+};
+
+type DataRow = {
+  id: string;
+  'transaction.duration': number;
+};
+
+const COLUMN_ORDER = [
```
Comment (DominikB2014) [inline static/app/views/starfish/components/sampleEvents.tsx:22]: Small nit, but you can type this nicely
```suggestion
type Keys = 'id' | 'transaction.duration'
type TableColumnHeader = GridColumnHeader<Keys>;


const COLUMN_ORDER: TableColumnHeader[] = [
```

File: static/app/views/starfish/components/sampleEvents.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      },
+    ])
+    .withSorts([
+      {
+        field: 'transaction.duration',
+        kind: 'desc',
+      },
+    ]);
+
+  function renderBodyCell(column: GridColumnHeader, row: DataRow): React.ReactNode {
```
Comment (DominikB2014) [inline static/app/views/starfish/components/sampleEvents.tsx:54]: If you go with the previous suggestion
```suggestion
  function renderBodyCell(column: TableColumnHeader, row: DataRow): React.ReactNode {
```

**Review by DominikB2014 — APPROVED** [review]
lgtm overall!

**Review by shruthilayaj — APPROVED** [review]

File: static/app/views/starfish/components/sampleEvents.tsx (hunk 1 above)
Comment (gggritso) [reply static/app/views/starfish/components/sampleEvents.tsx]: Typo, fixed

File: static/app/views/starfish/components/sampleEvents.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/starfish/components/sampleEvents.tsx]: 👌🏻

**Pull request merged**